docs/data/data-grid/filtering-recipes/filtering-recipes.md

```diff
@@ -13,3 +13,7 @@ Currently if you want to use the [Quick filter](/x/react-data-grid/filtering/qui
 A common use case is to have certain components positioned outside of the grid. Because of the way the grid context works this might not be a straightforward thing to do. The example below illustrates how this use case can be achieved.
 
 {{"demo": "QuickFilterOutsideOfGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Calculating row count in advance
+
+{{"demo": "FilteredRowCount.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
Hey @cherniavskii 
I am no allowed to push to your branch, so I'll just leave a suggestion here that you can apply

```suggestion

## Calculating row count in advance

In many data exploration scenarios, it's highly beneficial to display the row count associated with a predefined filter upfront.
This transparency empowers users to make informed decisions about which filter to apply, potentially saving them time and effort.

The [Grid API](/x/react-data-grid/api-object/#how-to-use-the-api-object) provides the [`getFilterState`](/x/api/data-grid/grid-api/#grid-api-prop-getFilterState) function, which allows you to retrieve the filter state for a given filter model without applying it to the data grid. By effectively utilizing `getFilterState`, you can dynamically calculate the row count that would result from applying a specific predefined filter.

{{"demo": "FilteredRowCount.js", "bg": "inline", "defaultCodeOpen": false}}
```

Reply:
@cherniavskii gentle ping 🔔